core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.java

```diff
@@ -80,6 +82,7 @@ private ClientOverrideConfiguration(Builder builder) {
         this.metricPublishers = Collections.unmodifiableList(new ArrayList<>(builder.metricPublishers()));
         this.executionAttributes = ExecutionAttributes.unmodifiableExecutionAttributes(builder.executionAttributes());
         this.scheduledExecutorService = builder.scheduledExecutorService();
+        this.requestCompressionConfiguration = builder.requestCompressionConfiguration();
```

Comment:
Did we add profile config option for `requestCompressionConfiguration`? I don't see it in this PR. Side note: it'd be great if we could create separate PRs for different features to make reviews eaiser, for example, separate PRs for codegen update and config options.